src/deepsparse/server/build_logger.py

```diff
@@ -0,0 +1,125 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Specifies the mapping from the ServerConfig to the DeepSparse Logger
+"""
+
+from typing import Any, Dict, List, Union
+
+from deepsparse import loggers as logger_objects
+from deepsparse.loggers.config import MetricFunctionConfig
+from deepsparse.server.config import ServerConfig
+
+
+__all__ = ["build_logger"]
+
+_LOGGER_MAPPING = {"python": logger_objects.PythonLogger}
+
+
+def build_logger(server_config: ServerConfig) -> logger_objects.BaseLogger:
```

Comment:
Have you tried using this in server code? I think a more natural formulation will be to create a logger for each individual endpoint using leaf loggers
```python
leaf_loggers = ...

def build_endpoint(endpoint_cfg: EndpointConfig, leaf_loggers: Dict[str, BaseLogger]) -> Pipeline:
    func_loggers = []
    for target_logging_cfg in endpoint_cfg.data_logging:
          target = leaf_loggers[target_logging_cfg.target]
          logger = FunctionLogger(logger=target, function=parse_function(...))
          func_loggers.append(logger)
    return Pipeline.create(..., logger=MultiLogger(func_loggers)
```

Reply:
I think the most important part for this PR is to sketch out the use in server code - TBH the actual implementation of build_logger can be left to another PR. Otherwise when we start trying to integrate into server we may run into things that require changes to build_logger - which would mean we double the effort. Instead we should sketch things out from top down and fill in internal implementation details after the full system is sketched out

Reply:
server integration will occur in a follow up PR since that increases the testing scope by quite a bit. 

each endpoint or each function? the flow under the hood can be refactored later if there is a use case.